nums/core/array/blockarray.py

```diff
@@ -445,6 +445,9 @@ def tensordot(self, other, axes=2):
         if not isinstance(other, BlockArray):
             raise ValueError("Cannot automatically construct BlockArray for tensor operations.")
 
+        assert isinstance(axes, int)
```

Comment:
Can you test this using an `if` statement, and raise an appropriate error if the test fails?